appinventor/appengine/src/com/google/appinventor/client/explorer/folder/FolderManager.java

```diff
@@ -59,7 +58,7 @@ public void loadFolders() {
     }
 
     LOG.info("folderJSON - " + folderJSON);
-    globalFolder = new ProjectFolder(folderJSON, null);
+    globalFolder = Ode.getUiFactory().createProjectFolder(folderJSON, null);
```

Comment:
I'd like to consider using dependency injection in classes like this rather than coupling them even more tightly to Ode. I.e., we should have the constructor take the UIFactory object and store it for references like here and elsewhere in the file. Ode, when it instantiates the folder manager, can pass the factory along. The field can be `final`.

Reply:
> I'd like to consider using dependency injection in classes like this rather than coupling them even more tightly to Ode. I.e., we should have the constructor take the UIFactory object and store it for references like here and elsewhere in the file. Ode, when it instantiates the folder manager, can pass the factory along. The field can be `final`.

OK. I think I understand what you're suggesting here, but I'd like to review it with you.